src/convnets/resnets.jl

```diff
@@ -0,0 +1,543 @@
+
+"""
+    basicblock(inplanes, planes; stride = 1, downsample = identity, cardinality = 1,
+               base_width = 64, reduce_first = 1, dilation = 1,
+               first_dilation = nothing, activation = relu, norm_layer = BatchNorm,
+               drop_block = identity, drop_path = identity,
+               attn_fn = planes -> identity, attn_args::NamedTuple = NamedTuple())
+
+Creates a basic ResNet block.
+
+# Arguments
+
+  - `inplanes`: number of input feature maps
+  - `planes`: number of feature maps for the block
+  - `stride`: the stride of the block
+  - `downsample`: the downsampling function to use
+  - `cardinality`: redundant, kept for compatibility with `bottleneck`.
+  - `base_width`: redundant, kept for compatibility with `bottleneck`.
+  - `reduce_first`: the reduction factor that the input feature maps are reduced by before the first
+    convolution.
+  - `dilation`: the dilation of the second convolution.
+  - `first_dilation`: the dilation of the first convolution.
+  - `activation`: the activation function to use.
+  - `norm_layer`: the normalization layer to use.
+  - `drop_block`: the drop block layer. This is usually initialised in the `_make_blocks`
+    function and passed in.
+  - `drop_path`: the drop path layer. This is usually initialised in the `_make_blocks`
+    function and passed in.
+  - `attn_fn`: the attention function to use. See [`squeeze_excite`](#) for an example.
+  - `attn_args`: a NamedTuple that contains none, some or all of the arguments to be passed to the
+    attention function.
+"""
+function basicblock(inplanes, planes; stride = 1, downsample = identity, cardinality = 1,
+                    base_width = 64, reduce_first = 1, dilation = 1,
+                    first_dilation = dilation, activation = relu, norm_layer = BatchNorm,
+                    drop_block = identity, drop_path = identity,
+                    attn_fn = planes -> identity, attn_args::NamedTuple = NamedTuple())
+    expansion = expansion_factor(basicblock)
+    @assert cardinality==1 "`basicblock` only supports cardinality of 1"
+    @assert base_width==64 "`basicblock` does not support changing base width"
+    first_planes = planes ÷ reduce_first
+    outplanes = planes * expansion
+    conv_bn1 = Chain(Conv((3, 3), inplanes => first_planes; stride, pad = first_dilation,
+                          dilation = first_dilation, bias = false),
+                     norm_layer(first_planes))
+    drop_block = drop_block
+    conv_bn2 = Chain(Conv((3, 3), first_planes => outplanes; pad = dilation,
+                          dilation = dilation, bias = false),
+                     norm_layer(outplanes))
+    attn_layer = attn_fn(outplanes; attn_args...)
+    return Chain(Parallel(+, downsample,
+                          Chain(conv_bn1, drop_block, activation, conv_bn2, attn_layer,
+                                drop_path)),
+                 activation)
```

Comment:
Why break apart the `+` and `activation`? The old API combined these into a single `connection` argument which allowed users to activate the branches vs. the combination as per their needs. This ties into why `addrelu` and `reluadd` need to be deleted? Makes sense with this change, but I think this change might reduce the flexibility?

Reply:
This is mostly because almost all the implementations I found do it this way. We could revert back to `addrelu` and `reluadd` but I don't think it makes a significant different in the accuracy or behaviour of the network

Reply:
Well I added that feature in order to use it 😅 so it'd be nice to keep it. As I recall, in that setting, I was working with a very small network with data pruning where it did make a difference for my baseline comparisons. Either way, it's more flexible to keep it as well as avoids removing an existing feature.

Reply:
That makes sense, it's easy enough to add it back 👍🏽